### ISSUE TITLE
Question: How do you handle Shopify product/page updates for static site deployment?

### ISSUE THREAD

**Issue by adamdehaven** [Gomah/nuxt-shopify#550]
I've run into a dilemma and I was wondering if you found a solution:

When using `nuxt-shopify` to generate and deploy a static site, the product information (and in my case, store pages) are fetched on the server during the build, and then hard-coded into the static site (working properly). This is perfect until someone updates one of the Shopify pages or product info, and now the data is out of date. I originally looked into building a Universal app, but for my use case, a static Nuxt site is currently the best solution.

How do you have the static content (so that it's available before hydration), but also update the site content with any new Shopify changes that occurred since the build?

Here's what I've done:

1. I fetch all content (products and pages) in the store via the `nuxtServerInit` method, which sets up the data for when the first page is loaded. (This is great assuming no changes have been made on Shopify between the build time and when the user accesses the page.)
2. When the user navigates to a product or page, I trigger a fetch event to go out and retrieve all products and pages again from the API, but on an interval (e.g. I only allow a fetch every 5 minutes).

This works; however, on the initial page load, there will be a flash of content if anything has changed. For example, if I change a product title, it will show incorrectly until the `$shopify` method returns the fresh data and writes it to the store. Then for the rest of the user's session, the newest data is shown.

I know I could show the loader or hide content until after the fetch completes; however, this would mean that no content is likely visible for SEO purposes, and the user would have to wait for products, images, etc. to load after they are fetched (which defeats the purpose of the static site, since there would originally not be an API call here).

I found a way around this by trigging a build via webhook on the host server any time certain Shopify events occur (e.g. product creation/deletion/update) that completely rebuilds and re-deploys the site. This solves my issue, but that could potentially be a lot of build time, which is less than ideal.

Any ideas? If you have the time to help me take a look, I have a project all set up.

**Comment by riccardolardi** [comment]
@adamdehaven any news on how you went on with this? I'm also planning to do a static deployment instead of SSR.

**Comment by adamdehaven** [comment]
@riccardolardi essentially I set `fetchOnServer` to false, and then manually grab the info on page load. Then, set up a Shopify webhook to hit your server (e.g. Netlify) to rebuild whenever you add/update/delete a product, etc. There isn't really a better solution.

You're more than welcome to check out the project I set up to play around: [@adamdehaven/shopify-nuxt](https://github.com/adamdehaven/shopify-nuxt)
